hs_rest_api/resources/file_metadata.py

```diff
@@ -0,0 +1,78 @@
+import json
+
+from rest_framework.response import Response
+from rest_framework import generics
+from rest_framework import serializers
+
+from hs_core.models import ResourceFile, Coverage
+
+
+# TODO: Once we upgrade past Django Rest Framework 3.3, this won't be necessary
+class JSONSerializerField(serializers.Field):
+    """ Serializer for JSONField -- required to make field writable"""
+
+    def to_internal_value(self, data):
+        return data
+    def to_representation(self, value):
+        return value
+
+
+class FileMetaDataSerializer(serializers.Serializer):
+    title = serializers.CharField()
+    keywords = JSONSerializerField()
+    spatial_coverage = JSONSerializerField()
+    extra_metadata = JSONSerializerField()
+    temporal_coverage = JSONSerializerField()
+
+
+class FileMetaDataRetrieveUpdateDestroy(generics.RetrieveUpdateDestroyAPIView):
+    serializer_class = FileMetaDataSerializer
+    allowed_methods = ('GET', 'PUT')
+
+    def get(self, request, pk, file_id):
+        """ Get a resource file's metadata. """
+        resource_file = ResourceFile.objects.get(id=file_id)
+        file_metadata = resource_file.metadata
```

Comment:
ResourceFile has no attribute ```metadata```. One has to navigate this way to get to file type metadata:
```file_metadata = resource_file.logical_file.metadata```

Not all resource files will have access to ```logical_file```. Therefore, one has to first check if a resource file is part of a logical file, if not then return 400.

```
if not resource_file.has_logical_file:
      return status.400_bad_request
 ```

Reply:
Odd - have you run the code? It totally works with `resource_file.metadata.` Why?

Reply:
I can add the 400 code but I want to get to the bottom of why my code works wihout `logical_file`

Reply:
Note ```resource_file.metadata``` can be ```None```

Reply:
I added a validator for both cases in the latest commit